x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -302,7 +302,7 @@ public void authorizeIndexAction(RequestInfo requestInfo, AuthorizationInfo auth
                 }
             }, listener::onFailure));
         } else {
-            authorizeIndexActionName(action, authorizationInfo, IndicesAccessControl.ALLOW_NO_INDICES,
+            authorizeIndexActionName(action, authorizationInfo, null,
```

Comment:
The `IndexAccessControl` parameter of the `authorizeIndexActionName` should be removed, I agree. But this is not something I'm going to take on in this PR because I believe it is an unwarranted detour. I also don't believe documentation is a proper fix in this case. The method is poorly structured but is very small and I always prefer to read the code to the docs/comments.

Reply:
I proposed two things here and maybe I should have kept them separate. 
* The first one is about the code change, i.e. `ALLOW_NO_INDICES` to `null`. The documentation suggestion is about this change. For me, it is hard to understand by just reading the code to see why this change is made and why it works. I had to run the code to understand that, for this particular case, it does *not* matter what value gets passed to the method since it is never really used. Also this change is an exception for this PR which mainly adds testing time behaviour. Therefore I think some docs are helpful.

* The second one is about the `authorizeIndexActionName` method, which I agree is out of scope for the current PR as I mentioned previously. The documentation was *not* suggested here. I meant refactoring of the method. But again I understand this is out of scope.

Reply:
Reverted back from `null` to `IndicesAccessControl.ALLOW_NO_INDICES`.